### PR TITLE
Defining and Instantiating Structs

### LINKED ISSUE #23: Defining and Instantiating Structs
- how to define structs
- how to mutate struct fields
- how to use as function return types
- using the field init shorthand
- unit-like structs without any fields

Estimated: 1 hour

### PR DESCRIPTION
**Pull request by trangnv**
Closes #23 
Time spend: 1 hour

Some notes:
- Assessing Ownership of Struct Data with lifetimes feature as TODO as I'm not familiar with Cairo *lifetimes* feature and this related to later chapter
- Code tested
- I presume we'll have chapter 3 for Ownership so I count this as chapter 4, for naming the Listings
- Please correct me if I'm wrong here: **Cairo doesn't have these features for structs**:
  - [Creating Instances from Other Instances with Struct Update Syntax](https://doc.rust-lang.org/book/ch05-01-defining-structs.html#creating-instances-from-other-instances-with-struct-update-syntax)
  - [Using Tuple Structs Without Named Fields to Create Different Types](https://doc.rust-lang.org/book/ch05-01-defining-structs.html#using-tuple-structs-without-named-fields-to-create-different-types)
  - [Unit-Like Structs Without Any Fields](https://doc.rust-lang.org/book/ch05-01-defining-structs.html#unit-like-structs-without-any-fields)

### CODE REVIEW

File: src/ch04-01-defining-and-instantiating-structs.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
+# Defining and Instantiating Structs
+
+Structs are similar to tuples, discussed in [The Data Types](ch02-02-data-types.md) section, in that both hold multiple related values. Like tuples, the pieces of a struct can be different types. Unlike with tuples, in a struct you’ll name each piece of data so it’s clear what the values mean. Adding these names means that structs are more flexible than tuples: you don’t have to rely on the order of the data to specify or access the values of an instance.
+
+To define a struct, we enter the keyword struct and name the entire struct. A struct’s name should describe the significance of the pieces of data being grouped together. Then, inside curly brackets, we define the names and types of the pieces of data, which we call fields. For example, Listing 4-1 shows a struct that stores information about a user account.
```
Comment (enitrat) [inline src/ch04-01-defining-and-instantiating-structs.md:5]: > the keyword struct

Formatting: the keyword `struct`

File: src/ch04-01-defining-and-instantiating-structs.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        username: 'someusername123',
+        email: 'someone@example.com',
+        sign_in_count: 1_u64,
+    };
+}
+```
+
+<span class="caption">Listing 4-2: Creating an instance of the `User` struct</span>
+
+To get a specific value from a struct, we use dot notation. For example, to access this user’s email address, we use `user1.email`
```
Comment (enitrat) [inline src/ch04-01-defining-and-instantiating-structs.md:48]: missing end-of-line dot. you can also merge this paragraph with the next one

**Review by enitrat — COMMENTED** [review]
Amazing PR - thank you, Trang !

**Comment by enitrat** [comment]
> Assessing Ownership of Struct Data with lifetimes feature as TODO as I'm not familiar with Cairo lifetimes feature and this related to later chapter

There's no such thing as lifetime, you can skip this part entirely :)

**Comment by trangnv** [comment]
Ready to be merged

**Review by enitrat — APPROVED** [review]

**Pull request merged**